### PR TITLE
Add Receive matcher

### PR DESCRIPTION
**Pull request by poy**
- Receive matcher reads from channels for assertions
- Add Receive func to To

### CODE REVIEW

**Comment by poy** [comment]
This would resolve issue #21

File: matchers/receive.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+		if !reflect.TypeOf(rxValue.Interface()).AssignableTo(outType.Elem()) {
+			return fmt.Errorf("can not assigned %s to %s",
+				reflect.TypeOf(rxValue.Interface()).String(),
+				reflect.TypeOf(m.ReceiveTo).String(),
+			)
+		}
+
+		outValue := reflect.ValueOf(m.ReceiveTo)
+		reflect.Indirect(outValue).Set(rxValue)
```
Comment (nelsam) [inline matchers/receive.go]: You should be able to do `outValue.Elem().Set(rxValue)` here.  `reflect.Indirect` has issues that `Elem()` doesn't.

File: matchers/receive.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	v := reflect.ValueOf(actual)
+	rxValue, ok := v.TryRecv()
+
+	if !ok {
+		return errors.New("did not receive")
+	}
+
+	if m.ReceiveTo != nil {
```
Comment (nelsam) [inline matchers/receive.go]: Can we just return nil if `m.ReceiveTo == nil`?

**Review by nelsam — CHANGES_REQUESTED** [review]
I'd also like to see `expect.Matcher` types allowed as the `ReceiveTo` field, some day.  It's not really an important first step, though.

**Comment by poy** [comment]
I made the proposed changes.

**Comment by poy** [comment]
I would like to first better understand how chaining will work before updating Receive to take a child matcher.

**Pull request closed without merging**